build.rs

```diff
@@ -0,0 +1,65 @@
+use std::{env, fs::File, io::prelude::*, path::PathBuf};
+
+fn main() {
+    if cfg!(feature = "ld") {
+        gen_memory_x();
+    }
+    println!("cargo:rerun-if-changed=build.rs");
+}
+
+fn gen_memory_x() {
+    #![allow(clippy::unneeded_wildcard_pattern)]
+
+    let mem_cfg_set = (
+        cfg!(feature = "mem-4"),
+        cfg!(feature = "mem-6"),
+        cfg!(feature = "mem-8"),
+        cfg!(feature = "mem-b"),
+        cfg!(feature = "mem-c"),
+        cfg!(feature = "mem-d"),
+        cfg!(feature = "mem-e"),
+    );
+    let flash = match mem_cfg_set {
+        (true, ..) => 16,
+        (_, true, ..) => 32,
+        (_, _, true, ..) => 64,
+        (.., true, _, _, _) => 128,
+        (.., _, true, _, _) => 256,
+        (.., _, _, true, _) => 384,
+        (.., _, _, _, true) => 512,
+        _ => unreachable!(),
```

Comment:
Replace with cleaner error handling, as discussed in https://github.com/stm32-rs/stm32f3xx-hal/pull/216#discussion_r620266515